enhancements/ingress/haproxy-reload-interval.md

```diff
@@ -0,0 +1,163 @@
+---
+title: haproxy-reload-interval
+authors:
+  - "@Ethany-RH"
+reviewers:
+  - "?"
+approvers:
+  - "?"
+api-approvers: # necessary?
+  - "?"
+creation-date: 2022-07-01
+last-updated: 2022-07-01
+tracking-link:
+  - "https://issues.redhat.com/browse/NE-586"
+see-also:
+replaces:
+superseded-by:
+---
+
+# Reload Interval in HAProxy
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`.
+- [ ] Design details are appropriately documented from clear requirements.
+- [ ] Test plan is defined.
+- [ ] graduation criteria for dev preview, tech preview, GA
+- [ ] User-Facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/).
+
+## Summary
+
+Add an API field to configure OpenShift router's `RELOAD_INTERVAL` environment variable so that administrators can define the minimum frequency the router is allowed to reload to accept new changes.
+
+OpenShift router currently hard-codes this reload interval to 5s. It should be possible for administrators to tune this value as necessary. Based on the processes run in the cluster and the frequency that it sees new changes, increasing the minimum interval at which the router is allowed to reload can improve its efficiency.
+This proposal extends the existing IngressController API to add a tuning option for the reload interval.
+
+## Motivation
+
+When there is an update to a route or endpoint in the cluster, the configuration for HAProxy changes, requiring that it reload for those changes to take effect. When HAProxy reloads to generate a new process with the updated configuration, it must keep the old process running until all its connections die. As such, frequent reloading increases the rate of accumulation of HAProxy processes, particularly if it has to handle many long-lived connections. The default reload interval is set as 5 seconds, which is too low for some scenarios, so it is important that administrators can extend this time interval.
+
+In addition, HAProxy's roundrobin balancing starts over from the first server every time HAProxy reloads. Thus, another motivating factor is that frequent reloads can cause load imbalance on a backend's servers when using the roundrobin balancing algorithm.
```

Comment:
This affects other balancing algorithms, not just roundrobin.  However, this is a good example—it's easy to explain how the issue affects roundrobin.